packages/dds/tree/src/feature-libraries/editable-tree/editableTree.ts

```diff
@@ -356,52 +374,61 @@ const handler: AdaptingProxyHandler<ProxyTarget, EditableTree> = {
             case proxyTargetSymbol: {
                 return target;
             }
-            case anchorSymbol: {
-                return target.getAnchor()
+            case insertNodeSymbol: {
+                return target.insertNode.bind(target);
+            }
+            case setValueSymbol: {
+                return target.setValue.bind(target);
+            }
+            case deleteNodeSymbol: {
+                return target.deleteNode.bind(target);
             }
             default:
                 return undefined;
         }
     },
-    set: (target: ProxyTarget, key: string | symbol, setValue: unknown, receiver: ProxyTarget): boolean => {
-        throw new Error("Not implemented.");
+    set: (target: ProxyTarget, key: string, value: unknown, receiver: unknown): boolean => {
+        if (target.has(key)) {
+            const typeName = target.getType(key, true) as TreeSchemaIdentifier;
+            return target.setValue(key, value, typeName);
+        }
+        const fieldSchema = getFieldSchema(target.getType() as TreeSchema, key);
+        assert(fieldSchema.types !== undefined && fieldSchema.types.size === 1,
+            "Cannot resolve a field type, use 'insertNodeSymbol' instead");
+        const name = [...fieldSchema.types][0];
+        const type: NamedTreeSchema = { name, ...target.context.forest.schema.lookupTreeSchema(name) };
+        const jsonValue = isPrimitiveValue(value) ? value : value as object;
```

Comment:
First I want to make sure I understand the editing contract that is implemented here: if I am following the code correctly, it looks like this adds support for three kinds of "set" operations:
1. Overwriting the primitive value of an existing node.
2. Inserting a new leaf node that represents the given primitive value where no such node exists currently.
3. Inserting a new subtree where no such node exists currently.
Is that correct?

Reply:
Yep, that's correct more or less. Currently it's limited with a functionality of `TypedJsonCursor`, e.g., one shouldn't use it if a schema contains polymorphic fields (same for the context of a subtree), or optional fields, etc.. Still a lot to be done here.

Reply:
You could also, in the case where there is a subtree present for the given key and the given `value` is not a primitive, replace the existing subtree with a combination of delete and insert.

I don't expect you to add that in this PR though, just pointing out the possibility.
Feel free to close this comment.

Reply:
I like the idea. I'll keep in mind.